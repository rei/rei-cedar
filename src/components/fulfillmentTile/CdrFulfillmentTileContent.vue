<script setup lang="ts">
import { useCssModule, computed } from 'vue';
import CdrBody, { type bodyTextProps } from '../text/presets/CdrBody.vue';
import { fulfillmentTileContent } from '../../types/interfaces';

/** Content component with scale option for fulfillment tile */

defineOptions({ name: 'CdrFulfillmentTileContent' });

const props = withDefaults(defineProps<fulfillmentTileContent>(), {
  stretch: false,
  scale: '-2',
});

const style = useCssModule();
const baseClass = 'cdr-fulfillment-tile-content';

const rootProps = computed(
  (): bodyTextProps => ({
    tag: 'div',
    scale: props.scale,
    class: {
      [style[baseClass]]: true,
      [style[`${baseClass}--stretch`]]: props.stretch,
    },
  }),
);
</script>

<template>
  <CdrBody v-bind="rootProps">
    <!-- @slot Where all default content should be placed. -->
    <slot />
  </CdrBody>
</template>

<style lang="scss" module src="./styles/CdrFulfillmentTileContent.module.scss"></style>
