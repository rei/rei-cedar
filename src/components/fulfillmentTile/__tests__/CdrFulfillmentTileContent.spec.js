import { mount } from '../../../../test/vue-jest-style-workaround.js';
import CdrFulfillmentTileContent from '../CdrFulfillmentTileContent.vue';

describe('CdrFulfillmentTileContent', () => {
  it('shows content', () => {
    const wrapper = mount(CdrFulfillmentTileContent, {
      slots: {
        default: 'Default slot',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.classes()).toEqual(['cdr-body', 'cdr-fulfillment-tile-content']);
    expect(wrapper.text()).toBe('Default slot');
  });

  it('is stretch', () => {
    const wrapper = mount(CdrFulfillmentTileContent, {
      props: {
        stretch: true,
      },
      slots: {
        default: 'Default slot',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.classes()).toEqual(['cdr-body', 'cdr-fulfillment-tile-content']);
  });
});
