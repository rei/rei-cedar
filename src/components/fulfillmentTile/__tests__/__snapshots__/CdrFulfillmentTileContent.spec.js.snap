// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CdrFulfillmentTileContent > is stretch 1`] = `
<div
  class="cdr-body cdr-fulfillment-tile-content"
  style="--cdr-body-font-size: var(--cdr-type-scale--2); --cdr-body-line-height: calc(var(--cdr-type-scale--2) * var(--cdr-body-line-height-ratio)); --cdr-body-font-weight: 400;"
>
  
  <!-- @slot Where all default content should be placed. -->
  
  Default slot
  
  
</div>
`;

exports[`CdrFulfillmentTileContent > shows content 1`] = `
<div
  class="cdr-body cdr-fulfillment-tile-content"
  style="--cdr-body-font-size: var(--cdr-type-scale--2); --cdr-body-line-height: calc(var(--cdr-type-scale--2) * var(--cdr-body-line-height-ratio)); --cdr-body-font-weight: 400;"
>
  
  <!-- @slot Where all default content should be placed. -->
  
  Default slot
  
  
</div>
`;
